test/integration/default/controls/_mapdata_spec.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+# Prepare platform "finger"
+platform_name = case platform[:name]
+                when 'amazon'
+                  'amazonlinux'
+                else
+                  platform[:name]
+                end
+
+platform_release = case platform[:name]
+                   when 'ubuntu'
+                     platform[:release].split('.').slice(0, 2).join
+                   when 'opensuse'
+                     "leap-#{platform[:release].split('.').slice(0, 2).join}"
+                   when 'arch'
+                     'base-latest'
+                   when 'amazon'
+                     # `2018` relase is named `1` in kitchen.yaml
+                     platform[:release].gsub(/2018.*/, '1')
+                   else
+                     platform[:release].split('.')[0]
+                   end
+
+platform_finger = "#{platform_name}-#{platform_release}"
```

Comment:
Considering that we're planning to reuse this in numerous formulas, I'm wondering if this isn't a bit too much (may cause contributors to feel a little overwhelmed).  In other formulas, I found that this was sufficient:

```suggestion
# Prepare platform "finger"
platform_finger = "#{platform[:name]}-#{platform[:release].split('.')[0]}"
```

Apart from the `amazonlinux` issue, what else would need to be adjusted here?

Reply:
This also gives `arch-5`, which can be adjusted as well.

Reply:
I'm planning to abstract this in `test/integration/share/libraries/system.rb` and provide a `system.platform` with grains like names and values:

- `system.platform[:os_family]` some of them are different between `salt` and `Inspec`
- `system.platfom[:osfinger]`
- ...

Reply:
Nice!

Reply:
My main concern here was to make things coherent between kitchen platform names and YAML names, that's why there is some quite amount of mangling.

But we could name the YAML files properly to only need:

```ruby
platform_finger = "#{platform[:name]}-#{platform[:release].split('.')[0]}"
```

Reply:
The complexity is now in the `system.rb` library.